api4/system.go

```diff
@@ -64,6 +68,150 @@ func (api *API) InitSystem() {
+	if !c.App.SessionHasPermissionToAny(*c.App.Session(), model.SysconsoleReadPermissions) {
+		c.SetPermissionError(model.SysconsoleReadPermissions...)
+		return
+	}
+
+	// Checking to see if the server has a e10 or e20 license (this feature is only permitted for servers with licenses)
+	if c.App.Srv().License() == nil {
+		c.Err = model.NewAppError("Api4.generateSupportPacket", "api.no_license", nil, "", http.StatusForbidden)
+		return
+	}
+
+	// Here we are getting information regarding Elastic Search
+	var elasticServerVersion string
+	var elasticServerPlugins []string
+	if c.App.SearchEngine().ElasticsearchEngine != nil {
+		elasticServerVersion = c.App.SearchEngine().ElasticsearchEngine.GetFullVersion()
+		elasticServerPlugins = c.App.SearchEngine().ElasticsearchEngine.GetPlugins()
+	}
+
+	// Here we are getting information regarding LDAP
+	ldapInterface := c.App.Ldap()
+	var vendorName, vendorVersion string
+	if ldapInterface != nil {
+		vendorName, vendorVersion = ldapInterface.GetVendorNameAndVendorVersion()
+	}
+
+	// Here we are getting information regarding the database (mysql/postgres + version)
+	databaseType, databaseVersion := c.App.Srv().DatabaseTypeAndVersion()
+
+	// Creating the struct for support packet yaml file
+	supportPacket := model.SupportPacket{
+		ServerOS:             runtime.GOOS,
+		ServerArchitecture:   runtime.GOARCH,
+		DatabaseType:         databaseType,
+		DatabaseVersion:      databaseVersion,
+		LdapVendorName:       vendorName,
+		LdapVendorVersion:    vendorVersion,
+		ElasticServerVersion: elasticServerVersion,
+		ElasticServerPlugins: elasticServerPlugins,
+	}
+
+	// Marshal to a Yaml File
+	supportPacketYaml, _ := yaml.Marshal(&supportPacket)
+
+	// Creating an array of files that we are going to be adding to our zip file
+	fileDatas := []model.FileData{
+		{
+			Filename: "support_packet.yaml",
+			Body:     supportPacketYaml,
+		},
+	}
+
+	// Getting the plugins installed on the server, prettify it, and then add them to the file data array
+	pluginsResponse, appErr := c.App.GetPlugins()
+	if appErr == nil {
+		pluginsPrettyJSON, err := json.MarshalIndent(pluginsResponse, "", "    ")
+		if err == nil {
+			fileDatas = append(fileDatas,
+				model.FileData{
+					Filename: "plugins.json",
+					Body:     pluginsPrettyJSON,
+				})
+		}
+	}
+
+	// Getting sanitized config, prettifying it, and then adding it to our file data array
+	sanitizedConfigPrettyJson, err := json.MarshalIndent(c.App.GetSanitizedConfig(), "", "    ")
+	if err == nil {
+		fileDatas = append(fileDatas,
+			model.FileData{
+				Filename: "sanitized_config.json",
+				Body:     sanitizedConfigPrettyJson,
+			})
+	}
+
+	// Getting mattermost.log and notifications.log
+	if *c.App.Srv().Config().LogSettings.EnableFile {
+		// mattermost.log
+		mattermostLog := utils.GetLogFileLocation(*c.App.Srv().Config().LogSettings.FileLocation)
+
+		mattermostLogFileData, mattermostLogFileDataErr := ioutil.ReadFile(mattermostLog)
```

Comment:
I guess I'm a little confused by this. So I did notice that in `utils/logger.go` we have a `LogRotateSize = 10000` which matches with what the front-end description says "It gets rotated every 10,000) but I couldn't ever find that variable being used in either the server repo or enterprise repo. 

I spent some time trying to figure out how the rotating log files works but really couldn't. Maybe we can setup a meeting to go over this together.

Reply:
I looked into this and I don't think we do. 
I noticed that we never set the rotating of the files by the lines (unless I missed something here) but rather by file size (https://github.com/mattermost/mattermost-server/blob/master/mlog/log.go#L141). We can see we are setting the Max Size of the file to 100mb (which means when the logger file reaches 100mb, it gets rotated). If there is several of these, the zip file size could become big pretty fast.

I think the extra work that needs to be done for this doesn't make sense. Having the current log (which can hold up to a 100mb of information) is enough I believe. Otherwise, we may get into situations where the zip file is too big to attach and may take a while to download.